lib/factories/ppt/slides.js

```diff
@@ -111,10 +110,10 @@ class SlideFactory {
         let slideKey = `ppt/slides/${slide.name}.xml`;
         let objectCount = 0;
 
```

Comment:
I think you could simplify this by creating a default object and letting `options` override it

```javascript
let defaults = Object.assign({
x: 0,
y: 591502,
cx: 9144000,
cy: 5674995
},options);
```
Also then there is no reason to use the variables `imageX`, `imageY` etc.

Reply:
You still have to do the unit conversions though.  So in the end I did:

```javascript
        options = Object.assign(
            {
                // defaults
                x: 100,
                y: 100,
                cx: 350,
                cy: 200,
            },
            options
        );

        options.x = PptxUnitHelper.fromPixels(options.x);
        options.y = PptxUnitHelper.fromPixels(options.y);
        options.cx = PptxUnitHelper.fromPixels(options.cx);
        options.cy = PptxUnitHelper.fromPixels(options.cy);
```

Reply:
I think we can simplify this logic some by making a helper method in this class that works something like this:

```javascript
assignDefaults(defaults = {}, options = {}) {
    Object.assign(defaults, options);
    options.x = PptxUnitHelper.fromPixels(options.x);
    options.y = PptxUnitHelper.fromPixels(options.y);
    options.cx = PptxUnitHelper.fromPixels(options.cx);
    options.cy = PptxUnitHelper.fromPixels(options.cy);
    return options;
}
```

Then you can call it with something like this:

```javascript
options = assignDefaults(
            {
                // defaults
                x: 100,
                y: 100,
                cx: 350,
                cy: 200,
            },
            options
        );
```

Reply:
Your first block isn't right. After options is assigned to defaults, you then need to pass _defaults_ into fromPixels() (because options is still the same at this point). But this makes the code unclear because now defaults might not really contain defaults anymore if options overrode some values. So what I did was this: create a redundant "settings" object that takes the return value from Object.assign(), and then use it inside fromPixels(), like:

```javascript
    static assignDefaults(defaults = {}, options = {}) {
        let settings = Object.assign(defaults, options);

        options.x = PptxUnitHelper.fromPixels(settings.x);
        options.y = PptxUnitHelper.fromPixels(settings.y);
        options.cx = PptxUnitHelper.fromPixels(settings.cx);
        options.cy = PptxUnitHelper.fromPixels(settings.cy);

        return options;
    }
```